util/lru_cache.cc

```diff
@@ -256,6 +256,19 @@ Cache::Handle* LRUCacheShard::Lookup(const Slice& key, uint32_t hash) {
   return reinterpret_cast<Cache::Handle*>(e);
 }
 
+bool LRUCacheShard::Ref(Cache::Handle* h) {
+  LRUHandle* handle = reinterpret_cast<LRUHandle*>(h);
+  MutexLock l(&mutex_);
+  if (handle->InCache()) {
+    if (handle->refs == 1) {
```

Comment:
But it could be greater than one if there are external references as well. refs == 1 should indicate the handle has no external refs and thus is it in the LRU list awaiting to be freed, in which case we need to remove it from the LRU list upon adding an external reference.